### DISCUSSION TITLE
10 characters name limit?

### DISCUSSION THREAD

**Discussion by evlo** [pvvx/ATC_MiThermometer#385] [Ideas]
I have quite a few of these devices and it is becoming a mess to manage the naming so I do not get lost i what is what.

I still think BT and especially these sensors are best option there is for battery operated TH sensors, but since I will be starting new project where tens of these will be used I was wondering if it would be possible to allow for more then 10 characters names.

I have seen some other devices broadcast long names, but I have no idea how that works.

example of the long name device on the image bellow 
<img src="https://github.com/pvvx/ATC_MiThermometer/assets/8813196/f04bf6bc-8911-4f5b-a259-ec21aa104cfc" width="30%" height="30%">

**Comment by evlo** [comment]
I think this is because of the limit of BLE advertisement message length being 31bytes (37 - MAC).
Honestly i did not look into all the protocols and messages devices actually support, but probably no reason to complicate things with option for longer device name in the advertisement messages.

There seem to be 0x09 Complete Local Name message type or actually use the current one as short name and add the new complete name - maybe that is what ie. quinging use to broadcast it's long name? I dunno how many people this even affects, but for me, with about 50 or so devices it is sometimes a challenge to figure out what is what so option for setting that "complete local name" would be appreciated,  if it is not complicated to implement.

**Comment by pvvx** [comment]
> I dunno how many people this even affects, but for me, with about 50 or so devices it is sometimes a challenge to figure out what is what so option for setting that "complete local name" would be appreciated, if it is not complicated to implement.

I only have more thermometers, and many of them are used for tests. I have no problems finding and determining where and what this device is. Even if you designate the use of a random MAC.

Likewise, many smart plugs from Tuya have the same BLE name.

**Comment by pvvx** [comment]
> I think this is because of the limit of BLE advertisement message length being 31bytes (37 - MAC).

These are the pre-2014 BLE standards. The following Bluetooth standards provide virtually unlimited length of advertising messages.

**Comment by evlo** [comment]
So um, can I just remove character limit on the name field on https://pvvx.github.io/ATC_MiThermometer/TelinkMiFlasher.html ?

**Comment by pvvx** [comment]
But in Linux we are limited by the incomplete implementation of the Bluetooth ver4.2 standard (2014).

> So um, can I just remove character limit on the name field on https://pvvx.github.io/ATC_MiThermometer/TelinkMiFlasher.html ?

No. This requires an increase in memory buffers in the program and increases consumption when transmitting more characters. Nobody needs this, since 10 characters is an extremely large number of combinations - such a number is not used in the real world.

If you want to increase the size of the buffer characters for the name, use the source code editor and rebuild your firmware.
https://github.com/pvvx/ATC_MiThermometer#building-the-firmware

**Comment by evlo** [comment]
My issue is not with uniquely identifying device, but having easily human readable names.

**Comment by pvvx** [comment]
> My issue is not with uniquely identifying device, but having easily human readable names.

If you want to increase the size of the buffer characters for the name, use the source code editor and rebuild your firmware.
Ubuntu:
```
  git clone https://github.com/pvvx/ATC_MiThermometer.git
  cd ATC_MiThermometer
  make
```

**Comment by AJolly** [comment]
FWIW, I too would like to have a longer name even if it's less efficient (without having to compile myself).  No worries if you don't want to add it though, I do really love and appreciate your firmware.

**Comment by pvvx** [comment]
The current version uses 18 bytes for the device name.
